nvtabular/loader/torch.py

```diff
@@ -113,10 +129,48 @@ def _LONG_DTYPE(self):
     def _FLOAT32_DTYPE(self):
         return torch.float32
 
-    def _handle_tensors(self, cats, conts, labels):
-        if isinstance(conts, torch.Tensor):
-            conts = conts.clone()
-        return cats, conts, labels
+    def _pull_values_offsets(self, values_offset):
+        # pull_values_offsets, return values offsets diff_offsets
+        if isinstance(values_offset, tuple):
+            values = values_offset[0].flatten()
+            offsets = values_offset[1].flatten()
+        else:
+            values = values_offset.flatten()
+            offsets = torch.from_numpy(np.arange(values.size()[0])).to("cuda")
+        num_rows = len(offsets)
+        offsets = torch.cat([offsets, torch.LongTensor([len(values)]).to("cuda")])
+        diff_offsets = offsets[1:] - offsets[:-1]
+        return values, offsets, diff_offsets, num_rows
+
+    def _get_max_seq_len(self, diff_offsets):
+        # get_max_seq_len, return int
+        return int(diff_offsets.max())
+
+        # Building the indices to reconstruct the sparse tensors
+
+    def _get_indices(self, offsets, diff_offsets):
+        row_ids = torch.arange(len(offsets) - 1).to("cuda")
```

Comment:
Can we create these directly on the GPU instead of creating on the CPU and then copying over with ```to("cuda")```?